### ISSUE TITLE
Illustration for website

### ISSUE THREAD

**Issue by rquey** [acmelab-ua/FEPX-website#3]
We want to have a cool illustration for the homepage. I was thinking of having a nice polycrystal on Neper's homepage, and an image of the same (but deformed) polycrystal on FEPX's homepage.
(maybe tessellation + mesh for Neper and mesh + result for FEPX)

The nice polycrystal could be:
- a standard, quite large grain-growth polycrystal
- a cylindrical dogbone sample (approximately)
- a 2-scale polycrystal

In any case, it is good to have an elongated sample, and a fine mesh...

Maybe we don't want to show something too involved w/ current capabilities (cylinders / dogbones are not simply supported). I wouldn't mind changing the Neper image to something simpler.

Thoughts?

**Comment by rquey** [comment]
This looks quite nice, but we have to think about how it's going to go with the images on the Neper website.

- It looks nice to have 2-3 images next to each other, so let's keep that
- Neper is about tessellation and mesh, so there should be at least these 2 images
- FEPX is about mesh and simulation, so there should be at least these 2 images <-- I would show elements here; to make it less apparent, it works well to color them slightly away from black, e.g. `16|16|16`.

I would like to try something as shown below to plot the crystals orientations and phases (sometimes used for far-field 3DXRD data). 
What do you think?

I'm playing with the script to adjust the figures (camera, etc.). I'll try to superimpose the 3 images directly in the pov file, too.
@mkasemer, can you share the coo and strain-eq files you used for the plot?

@mkasemer, there is some novelty in this plot on the Neper front (dual phase material), and it's the first FEPX release, so I think it's quite nice. Possibilities for improvements: bigger polycrystal / mesh (but we don't want too small grains), larger strain, more complex structure (2-scale).

![image](https://user-images.githubusercontent.com/2659376/87944859-3a4e2780-caa0-11ea-94d0-1ac434d77426.png)

**Comment by mkasemer** [comment]
Agreed on the general plan: Neper to show off tess/mesh, FEPX to show off mesh/sim (there can of course be a bit of overlap).

Open to generation parameters for a larger polycrystal / structure. If you want to attach an example, please do so.

As far as the orientation plot goes, it is very interesting. Would you be able to plot oriented hexagonal prisms for an HCP material as well (just curious!)?

If we show this orientation plot, it may be best to do so for a slightly texture material, and could we color the cubes based on orientation?

**Comment by mkasemer** [comment]
Looking good for now - would we add a 4th image of the strain field? I think this would be quite complete.

How do you get multiple images plotted at once? Currently @jcappola is stitching them in Gimp (I have little skill), but if this is avoidable... (is it a headache?).

I will let you iterate to what you think is perhaps a more interesting microstructure, and I will run everything on my end. As a note, I would say the through thickness of the sample could be increased slightly to allow for 2-3 grains across the cross section. I don't want to imply that we are a pseudo 3D code (yes I know we can see two grains across the cross section in the above example...).

**Comment by rquey** [comment]
For FEPX, I would start with the mesh. Then maybe you can plot both the strains and stresses to get 3 images total? Let me try with what I have.

To get the image, I plot the 2 first as `pov:objects` and include them in the 3rd (after some translations) with `-includepov`. So, it's all automated, and it's really 3 objects - you can actually see the mesh through the cubes...

Yes, I think a thicker sample is the way to go (x 2 ?). You can also decrease `val` to `1e-2` or even `1e-3` in `-morphooptistop`, which will separate modes more clearly.

**Comment by mkasemer** [comment]
mesh, stress, strain - this is good!

I'm uncertain what this would look like (would have to see it implemented in script...).

Currently we have:
```
N="from_morpho"
RCL=0.65
DOMAIN="cube(0.3,0.075,1.25)"
MORPHO="diameq:lognormal(0.035,0.01)+0.2*lognormal(0.075,0.02),1-sphericity:lognormal(0.145,0.03)"
...
-morphooptistop "val=1e-1,xreps=0.001"
```

Maybe I'll try the following?:
```
N="from_morpho"
RCL=*0.5*
DOMAIN="cube(0.3,*0.15*,1.25)"
MORPHO="diameq:lognormal(0.035,0.01)+0.2*lognormal(0.075,0.02),1-sphericity:lognormal(0.145,0.03)"
...
-morphooptistop "*val=1e-2*,xreps=0.001" \
-group "mode!=2?1:2" \
```

Any suggestions for possible changes to `MORPHO`?

**Comment by rquey** [comment]
All good. I think this `-morpho` works quite well.

**Comment by rquey** [comment]
@mkasemer, can you please attach `specimen.msh`, for the simulation you run?

**Comment by mkasemer** [comment]
Yes, I will upload the mesh when the simulation has completed successfully.

![specimen-tess](https://user-images.githubusercontent.com/2902614/87956938-3f49b180-ca75-11ea-8813-24b1bce82d31.png)
![specimen-phase](https://user-images.githubusercontent.com/2902614/87956933-3d7fee00-ca75-11ea-8634-7120e255177c.png)

**Comment by rquey** [comment]
@mkasemer, sorry, I meant for the simulation corresponding to the coo and strain-eq files you attached earlier. I don't have the same number of nodes/elements + you likely partitioned differently.

**Comment by mkasemer** [comment]
Yes, here is the previous mesh (had partitioned for HPC).

https://drive.google.com/file/d/1IXPoX3lb6H0zAK_xBECZo1Un86sO5GJX/view?usp=sharing

**Comment by mkasemer** [comment]
New mesh (finer, more grains, thicker sample)

![specimen-mesh-phase](https://user-images.githubusercontent.com/2902614/87959177-1840af00-ca78-11ea-9766-fbce7eb21756.png)
![specimen-mesh](https://user-images.githubusercontent.com/2902614/87959188-1b3b9f80-ca78-11ea-9294-78bed2ef2de7.png)

**Comment by rquey** [comment]
Looks good!

**Comment by mkasemer** [comment]
@rquey: I now have the new data and am processing images. If I want to plot stress33, how would I do this using your visualization command?

**Comment by rquey** [comment]
`-dataeltcol stress33`

**Comment by rquey** [comment]
@mkasemer, I reworked the script very much. Here it goes with images for the previous data set:

### Neper
![image](https://user-images.githubusercontent.com/2659376/88063345-37b80480-cb6a-11ea-8116-126def66a48f.png)

### FEPX
![image](https://user-images.githubusercontent.com/2659376/88063364-3e467c00-cb6a-11ea-8661-8d553878ba88.png)

### Neper/FEPX
![image](https://user-images.githubusercontent.com/2659376/88063386-43a3c680-cb6a-11ea-8d53-158e748132ad.png)

**Comment by mkasemer** [comment]
Looks good! I will get you the new mesh and results shortly - playing with limit bounds to try for the best images.

**Comment by mkasemer** [comment]
@rquey: Do you have the ability to post-process variables like stress-eq based on stress output? We had discussed this, but I'm not sure if it was ever implemented.

Right now, I'm plotting strain_eq and stress_33. Here are the images. I did not print stress-eq from the simulation. I'm not sure the stress_33 plot will be good to show right next to the strain_eq plot (or maybe so, it's arbitrary, really).

![specimen-mesh-phase](https://user-images.githubusercontent.com/2902614/88065309-0f141880-cb32-11ea-8312-133e97a1c81e.png)
![specimen-mesh](https://user-images.githubusercontent.com/2902614/88065342-19cead80-cb32-11ea-993d-7f3df2420ac4.png)
![strain-eq-step7](https://user-images.githubusercontent.com/2902614/88065931-cf99fc00-cb32-11ea-9e08-b0966ada0e4c.png)
![stress33-step7](https://user-images.githubusercontent.com/2902614/88065964-d9bbfa80-cb32-11ea-8961-9ad4af4c8ded.png)

**Comment by rquey** [comment]
> @rquey: Do you have the ability to post-process variables like stress-eq based on stress output? We had discussed this, but I'm not sure if it was ever implemented.

No.

**Comment by mkasemer** [comment]
I can re-run to print stress-eq if you think we should use that. Otherwise, I think this is okay for now.

Here is the link to the mesh and field variables:

https://drive.google.com/file/d/1qv2Pkj3xWYVUKKPVGP4hqi65aD63YGgx/view?usp=sharing

**Comment by rquey** [comment]
@mkasemer, When ready, can you send the tess, mesh, coo, strain-eq and stress files (and specify the limit bounds you used)? I'll probably need to adjust the script before I can send it to you.

**Comment by mkasemer** [comment]
Here is the new package of everything you requested.

https://drive.google.com/file/d/1pQNuE7xmSfzx74lJz8JyXzW3wBwPH3tA/view?usp=sharing

All plotting can be seen in `home.sh`.
For equivalent strain, bounds are `-dataeltscale 0.00:0.1`
For stress33, bounds are `-dataeltscale 200:900`

**Comment by rquey** [comment]
![image](https://user-images.githubusercontent.com/2659376/88081752-24fcfa00-cb81-11ea-8fb8-e6dea4825df2.png)
![image](https://user-images.githubusercontent.com/2659376/88081773-2dedcb80-cb81-11ea-8896-a70e25c58492.png)
![image](https://user-images.githubusercontent.com/2659376/88081792-334b1600-cb81-11ea-867b-b8266339f91d.png)

Still have to play with the camera, maybe...

**Comment by mkasemer** [comment]
My only comment would be to perhaps move the camera down a bit to reduce the perspective distortion (can you make the vertical edges more vertical). But otherwise, looks great!

**Comment by rquey** [comment]
That would be more a camera distance effect.  I changed the object positions + camera distance, position and angle to reduce the perspective effect and have the objects appearing more horizontally aligned.

Concerning colors, no, it won't be easy for cubes. Strictly speaking, the image defines it all: orientation with the oriented cube and phase with the color. But would you like different colors simply to get more more contrast between cubes, for visualization (which seems doable, maybe by randomizing the colors a bit), or would you like something quantitative?

![image](https://user-images.githubusercontent.com/2659376/88093924-0142af80-cb93-11ea-9baf-6fbb8b24076b.png)
![image](https://user-images.githubusercontent.com/2659376/88093939-06076380-cb93-11ea-990c-1fb156ee4490.png)
![image](https://user-images.githubusercontent.com/2659376/88093944-09025400-cb93-11ea-827d-aaaf7c1e8742.png)

**Comment by mkasemer** [comment]
I was thinking something quantitative, to relate the color of the cube to it's orientation (in conjunction with the physical orientation of the cube). Just thinking, though if this is difficult or messy, it is likely not worth the overhead.

I like this image! I will be putting this on the website shortly!

**Comment by rquey** [comment]
Here it is. It generates the images from `specimen.tess`, `simulation.msh`, `coo.step7`, `strain-eq.step7` and `stress.step7`. I've made small changes since yesterday.

[home.sh.txt](https://github.com/acmelab-ua/FEPX-website/files/4960235/home.sh.txt)

![image](https://user-images.githubusercontent.com/2659376/88182111-1588ba00-cc30-11ea-977c-15a76d199d23.png)
![image](https://user-images.githubusercontent.com/2659376/88182122-191c4100-cc30-11ea-9bb3-88465b6f577f.png)
![image](https://user-images.githubusercontent.com/2659376/88182134-1c173180-cc30-11ea-95b9-ff229b4910f7.png)

**Comment by rquey** [comment]
@mkasemer, I made small changes to `index.rst` to adjust the column widths, now that we have a new image. I also updated the title.

Pushed to `main`.

**Comment by mkasemer** [comment]
It's up!
